src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -44,8 +44,27 @@ protected TelemetryConfigurationFactory()
         /// </remarks>
         public static TelemetryConfigurationFactory Instance
         {
-            get { return instance ?? (instance = new TelemetryConfigurationFactory()); }
-            set { instance = value; }
+            // Both get and set should write something to CoreEventSource so as to initialize it. This is a fix for below deadlock issue   
```

Comment:
Is there some other place in SDK code that is guaranteed to be invoked before the instance of TelemetryConfigurationFactory is get/set? That would be a good place to log with Event Source and initialize it for the first time. Or TelemetryConfigurationFactory getter/setter is indeed such a place?